custom_components/ams/config_flow.py

```diff
@@ -34,8 +34,6 @@ def __init__(self):
         return self.async_show_form(step_id="user", data_schema=vol.Schema({
             vol.Required(CONF_SERIAL_PORT,
                          default=None): vol.In(ports),
-            vol.Required(CONF_METER_MANUFACTURER,
```

Comment:
I think we should keep this but make it optional and default to None and just figure out the manufacturer if it None.

Reply:
It is possible to have it optional for YAML config, but config_flow does not allow for None values or blank fields in the flowhandler. See https://github.com/home-assistant/core/issues/30381
So these are the options:
1: Remove the manual setup completely. (Always autodetect)
2: Optional configuration for the meter manufacturer. This will be manual in config_flow and automatic if none with YAML.

Reply:
We can set the default value to auto. So there are 4 options in the place holders. Auto uses the parser, what ever value uses the parser the user has choosen. Same goes for yaml.

Reply:
:+1: Done